odoo/addons/base/models/ir_model.py

```diff
@@ -746,39 +746,46 @@ def _reflect_field(self, field):
         fields_data = self._existing_field_data(field.model_name)
         field_data = fields_data.get(field.name)
         params = self._reflect_field_params(field)
+        cr = self.env.cr
 
         if field_data is None:
-            cr = self.env.cr
-            # create an entry in this table
+            # does not exist, create an entry in this table
             query_insert(cr, self._table, params)
             record = self.browse(cr.fetchone())
             self.pool.post_init(record.modified, list(params))
-            # create a corresponding xml id
-            module = field._module or self._context.get('module')
-            if module:
-                model_name = field.model_name.replace('.', '_')
-                xmlid = 'field_%s__%s' % (model_name, field.name)
-                cr.execute(""" INSERT INTO ir_model_data (module, name, model, res_id, date_init, date_update)
-                               VALUES (%s, %s, %s, %s, (now() at time zone 'UTC'), (now() at time zone 'UTC')) """,
-                           (module, xmlid, record._name, record.id))
             # update fields_data (for recursive calls)
             fields_data[field.name] = dict(params, id=record.id)
-            return record
-
-        diff = {key for key, val in params.items() if field_data[key] != val}
-        if diff:
-            cr = self.env.cr
-            # update the entry in this table
-            query_update(cr, self._table, params, ['model', 'name'])
-            record = self.browse(cr.fetchone())
-            self.pool.post_init(record.modified, diff)
-            # update fields_data (for recursive calls)
-            field_data.update(params)
-            return record
-
         else:
-            # nothing to update, simply return the corresponding record
-            return self.browse(field_data['id'])
+            # exists
+            diff = {key for key, val in params.items() if field_data[key] != val}
+            if diff:
+                # update the entry in this table
+                query_update(cr, self._table, params, ['model', 'name'])
+                record = self.browse(cr.fetchone())
+                self.pool.post_init(record.modified, diff)
+                # update fields_data (for recursive calls)
+                field_data.update(params)
+            else:
+                # nothing to update
+                record = self.browse(field_data['id'])
+
+        # generate xmlids if necessary, one per module defining the same field
```

Comment:
Here is suggest to create the missing xmlid for `self._context['module']` if present:
```python
if self._context.get('module') in field._modules:
    # check xmlid and create if not found
```